src/include/sof/trace/trace.h

```diff
@@ -335,7 +335,7 @@ struct tr_ctx {
  * @param default_log_level Default log level.
  */
 #define DECLARE_TR_CTX(ctx_name, uuid, default_log_level)	\
-	struct tr_ctx ctx_name TRACE_CONTEXT_SECTION = {	\
+	const struct tr_ctx ctx_name TRACE_CONTEXT_SECTION = {	\
 		.uuid_p = uuid,					\
 		.level = default_log_level,			\
```

Comment:
level is not constant btw, can you check, it changes with the log level IIRC

Reply:
@lgirdwood you're so right... `trace_filter_update_global()` changes `.level` in place, and the compiler doesn't catch that with my PR because individual trace context objects aren't changed themselves instead the whole section where they are located is scanned and there they are modified... Dropping this PR. Thanks for spotting this!

Reply:
Can we make these SHARED instead OR aligned on cacheline size ?